basilisk/src/ast/references.c

```
@@ -0,0 +1,333 @@
+	  break;
+      }
+      if (found)
+	stack_push (a->attributes, &found);
+    }
+  }
+
+  ast_pop_scope (stack, scope);
+}
+
+static
+char * add_reference (Ast * ref, char * references, Ast * scope, Stack * stack, Stack * functions)
```

Clang (14.0.6)

[warning] function 'add_reference' has cognitive complexity of 68 (threshold 25)

(readability-function-cognitive-complexity)